Add WFC3/UVIS G280 Transit Notebook

FDauphin opened the pull request:
This notebook checklist has been made available to us by the [Notebooks For All team](https://github.com/Iota-School/notebooks-for-all/blob/main/resources/event-hackathon/notebook-authoring-checklist.md).
Its purpose is to serve as a guide for both the notebook author and the technical reviewer highlighting critical aspects to consider when striving to develop an accessible and effective notebook.

### The First Cell

- [ ] The title of the notebook in a first-level heading (eg. `<h1>` or `# in markdown`).
- [ ] A brief description of the notebook.
- [ ] A table of contents in an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists) (`1., 2.,` etc. in Markdown).
- [ ] The author(s) and affiliation(s) (if relevant).
- [ ] The date first published.
- [ ] The date last edited (if relevant).
- [ ] A link to the notebook's source(s) (if relevant).

### The Rest of the Cells

- [ ] There is only one H1 (`#` in Markdown) used in the notebook.
- [ ] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order (meaning it does not skip numbers). 

## Text

- [ ] All link text is descriptive. It tells users where they will be taken if they open the link.
- [ ] All acronyms are defined at least the first time they are used. 
- [ ] Field-specific/specialized terms are used when needed, but not excessively.

## Code

- [ ] Code sections are introduced and explained before they appear in the notebook. This can be fulfilled with a heading in a prior Markdown cell, a sentence preceding it, or a code comment in the code section.
- [ ] Code has explanatory comments (if relevant). This is most important for long sections of code.
- [ ] If the author has control over the syntax highlighting theme in the notebook, that theme has enough color contrast to be legible.
- [ ] Code and code explanations focus on one task at a time. Unless comparison is the point of the notebook, only one method for completing the task is described at a time.

### Images

- [ ] All images (jpg, png, svgs) have an [image description](https://www.w3.org/WAI/tutorials/images/decision-tree/). This could be
    - [ ] [Alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) (an `alt` property) 
    - [ ] [Empty alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt#decorative_images) for decorative images/images meant to be skipped (an `alt` attribute with no value)
    - [ ] Captions
    - [ ] If no other options will work, the image is decribed in surrounding paragraphs.

- [ ] Any [text present in images](https://www.w3.org/WAI/WCAG21/Understanding/images-of-text.html) exists in a text form outside of the image (this can be alt text, captions, or surrounding text.)

### Visualizations

- [ ] All visualizations have an image description. Review the previous section, Images, for more information on how to add it.
- [ ] [Visualization descriptions](http://diagramcenter.org/specific-guidelines-e.html) include
    - [ ] The type of visualization (like bar chart, scatter plot, etc.)
    - [ ] Title
    - [ ] Axis labels and range
    - [ ] Key or legend
    - [ ] An explanation of the visualization's significance to the notebook (like the trend, an outlier in the data, what the author learned from it, etc.)

- [ ] All visualizations and their parts have [enough color contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) ([color contrast checker](https://webaim.org/resources/contrastchecker/)) to be legible. Remember that transparent colors have lower contrast than their opaque versions.
- [ ] All visualizations [convey information with more visual cues than color coding](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html). Use text labels, patterns, or icons alongside color to achieve this.
- [ ] All visualizations have an additional way for notebook readers to access the information. Linking to the original data, including a table of the data in the same notebook, or sonifying the plot are all options.

FDauphin commented:
This PR is for adding a new WFC3 notebook for G280 exoplanet transits. It contains:
- a new folder `uvis_g280_transit`
  - The notebook `G280_Exoplanet_Transits.ipynb`
  - The complementary module `g280_transit_tools.py`
  - `pre-requirements.sh` which git clones [this repo ](https://github.com/npirzkal/GRISM_WFC3)needed in the notebook
  - `requirements.txt`
  - `G280sky_chip2_flt.fits` which is a G280 sky frame. The current link for downloading is broken so we are temporarily adding until it gets fixed
- The new notebook was added to the WFC3 `README.md`, WFC3 `general_tools.md`, and the Table of Contents (`_toc.yml`)

FDauphin commented:
@cshanahan1 @gibsongreen The notebook is failing an execution and validation test with this error:
```
FileNotFoundError: [Errno 2] No such file or directory: 'GRISM_WFC3/UVIS/UVIS_G280_CCD2_V2.conf'
```

That file is in [GRISM_WFC3](https://github.com/npirzkal/GRISM_WFC3), which should be cloned as specified in `pre-requirements.sh`. How do we confirm that this repository is being cloned properly?

FDauphin commented:
@cshanahan1 @gibsongreen I removed `pre-requirements.sh` and added a coding block in the notebook to clone the repo. This seemed appropriate since it's a folder of files and doesn't need any further installation. I also removed `G280sky_chip2_flt.fits`, and added another coding block in the notebook to download the file instead. 

With these changes, the notebook passed all CI checks and is ready for a technical review. Thanks in advance 🙌🏾

cshanahan1 commented:
thanks @FDauphin! will be reviewing this shortly

cshanahan1 reviewed (changes requested):
This looks great! Left a couple of minor style fix comments on the notebook, but otherwise everything looks great

FDauphin commented on notebooks/WFC3/uvis_g280_transit/G280_Exoplanet_Transits.ipynb:
```diff
@@ -0,0 +1,1309 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for column in sources.colnames:    \n",
```
Done

FDauphin commented on notebooks/WFC3/uvis_g280_transit/G280_Exoplanet_Transits.ipynb:
```diff
@@ -0,0 +1,1309 @@
+    "\n",
+    "Now, we remove cosmic rays, which can not only hit the detector throughout the course of the hours-long time-series observations, but may also be more frequent in G280 data due to its wide wavelength coverage. It is important to correct for cosmic rays within an exposure (spatially) and also across exposures in the time-series (temporally).\n",
+    "\n",
+    "Firest, we apply the temporal and spatial cosmic ray correction routines using `g280_transit_tools.remove_cosmic_rays_time` and `g280_transit_tools.remove_cosmic_rays_space`. These functions return the corrected data, and the masks for corrected pixels."
```
Done

FDauphin commented on notebooks/WFC3/uvis_g280_transit/G280_Exoplanet_Transits.ipynb:
```diff
@@ -0,0 +1,1309 @@
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```
Done

FDauphin commented:
@cshanahan1 Thanks for the speedy review! Sorry for the delay; got back from vacation. I addressed all your comments. Please let me know if there's anything else to do before merging 👍🏾
